database/database.go

```diff
@@ -163,18 +163,37 @@ func (s *DatabaseService) GetBuilderProfits(since, until time.Time) (res []*Buil
 	startSlot := timeToSlot(since)
 	endSlot := timeToSlot(until)
 
-	query := `SELECT
-		extra_data,
-		count(extra_data) as blocks,
-		count(extra_data) filter (where coinbase_diff_eth > 0) as blocks_profit,
-		count(extra_data) filter (where coinbase_diff_eth < 0) as blocks_sub,
-		round(avg(CASE WHEN coinbase_diff_eth IS NOT NULL THEN coinbase_diff_eth ELSE 0 END), 4) as avg_profit_per_block,
-		round(PERCENTILE_DISC(0.5) WITHIN GROUP(ORDER BY CASE WHEN coinbase_diff_eth IS NOT NULL THEN coinbase_diff_eth ELSE 0 END), 4) as median_profit_per_block,
-		round(sum(CASE WHEN coinbase_diff_eth IS NOT NULL THEN coinbase_diff_eth ELSE 0 END), 4) as total_profit,
-		round(abs(sum(CASE WHEN coinbase_diff_eth < 0 THEN coinbase_diff_eth ELSE 0 END)), 4) as total_subsidies
-	FROM (
-		SELECT distinct(slot), extra_data, coinbase_diff_eth FROM ` + vars.TableDataAPIPayloadDelivered + ` WHERE value_check_ok IS NOT NULL AND slot >= $1 AND slot <= $2
-	) AS x
+	query := `WITH 
+		payloads as (
+      SELECT 
+        distinct(slot), extra_data, coinbase_diff_eth, value_delivered_eth, block_hash, builder_pubkey
+      FROM ` + vars.TableDataAPIPayloadDelivered + ` WHERE value_check_ok IS NOT NULL AND slot >= $1 AND slot <= $2
+    )
+
+    , bid_adjustments as (SELECT slot, adjusted_value, builder_pubkey, adjusted_block_hash FROM ` + vars.TableAdjustments + ` WHERE slot >= $1 AND slot <= $2)
+
+    , adjusted_payloads as (
+      select
+        p.slot,
+        p.extra_data,
+        CASE
+          WHEN a.slot IS NOT NULL THEN p.coinbase_diff_eth - p.value_delivered_eth 
+          ELSE p.coinbase_diff_eth
+        END as coinbase_diff_eth
+      FROM payloads p
+      LEFT JOIN bid_adjustments a ON p.slot=a.slot AND p.builder_pubkey = a.builder_pubkey AND p.block_hash = a.adjusted_block_hash
+    )
+
+    SELECT
+      extra_data,
+      count(extra_data) as blocks,
+      count(extra_data) filter (where coinbase_diff_eth > 0) as blocks_profit,
+      count(extra_data) filter (where coinbase_diff_eth < 0) as blocks_sub,
+      round(avg(CASE WHEN coinbase_diff_eth IS NOT NULL THEN coinbase_diff_eth ELSE 0 END), 4) as avg_profit_per_block,
+      round(PERCENTILE_DISC(0.5) WITHIN GROUP(ORDER BY CASE WHEN coinbase_diff_eth IS NOT NULL THEN coinbase_diff_eth ELSE 0 END), 4) as median_profit_per_block,
+      round(sum(CASE WHEN coinbase_diff_eth IS NOT NULL THEN coinbase_diff_eth ELSE 0 END), 4) as total_profit,
+      round(abs(sum(CASE WHEN coinbase_diff_eth < 0 THEN coinbase_diff_eth ELSE 0 END)), 4) as total_subsidies
+    FROM adjusted_payloads
```

Comment:
This query worries me a little bit -- how much more expensive is it to run? Not sure if this could be a bottleneck. Do you have any data or estimates on query performance?